lib/rules/no-extra-boolean-cast.js

```diff
@@ -91,7 +97,14 @@ module.exports = {
                     context.report({
                         node,
                         message: "Redundant Boolean call.",
-                        fix: fixer => fixer.replaceText(node, sourceCode.getText(node.arguments[0]))
+                        fix: fixer => {
+                            const argument = node.arguments[0];
+
+                            if (astUtils.getPrecedence(argument) < astUtils.getPrecedence(node)) {
```

Comment:
I'm not 100% sure if I get you suggestion right @not-an-aardvark . In this case node precedence will always be the same because its CallExpression every time.